### PR TITLE
Feat/scan reward

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview
- Add scan QR tab scan reward
- Show list ticket rewards able to scan
- Show reward remaining information
- On tap toggle claim reward

## Demo

https://github.com/lemonadesocial/lemonade-flutter/assets/7247063/ee8b7734-5cf2-4c8b-a978-85cc92044769

### CODE REVIEW

File: lib/core/application/event/claim_rewards_bloc/claim_rewards_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/domain/user/user_repository.dart';
+import 'package:app/graphql/backend/schema.graphql.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'claim_rewards_bloc.freezed.dart';
+
+class ClaimRewardsBloc extends Bloc<ClaimRewardsEvent, ClaimRewardsState> {
```
Comment (mthinh) [inline lib/core/application/event/claim_rewards_bloc/claim_rewards_bloc.dart:12]: This bloc obviously use for getting rewards, should we rename it ? "Claim" is another action

File: lib/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/claim_rewards_page.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+    String eventId = context.watch<GetEventDetailBloc>().state.maybeWhen(
+              fetched: (event) => event.id,
+              orElse: () => "",
+            ) ??
+        "";
+    return MultiBlocProvider(
```
Comment (mthinh) [inline lib/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/claim_rewards_page.dart:31]: I saw only 1 bloc, dont need to use MultiBlocProvider

File: lib/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/claim_rewards_page.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      providers: [
+        BlocProvider(
```
Comment (mthinh) [inline lib/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/claim_rewards_page.dart:33]: can we move up BlocProvider 1 widget tree level ? 
ClaimRewardPage + BlocProvider => ClaimRewardPageView

File: lib/core/presentation/pages/scan_qr_code/views/scan_qr_code_view.dart [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import 'package:app/core/utils/snackbar_utils.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/router/app_router.gr.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+
+class ScanQRCodeView extends StatefulWidget {
```
Comment (mthinh) [inline lib/core/presentation/pages/scan_qr_code/views/scan_qr_code_view.dart:16]: This page seems couple with reward feature, if you cannot do it generic, then lets rename with specific use case , and move to event setting rewards page

**Comment by sieuhuflit** [comment]
Resolved + self test again, please help recheck 🙏

**Review by mthinh — APPROVED** [review]

**Pull request merged**